liblangutil/Scanner.cpp

```diff
@@ -826,31 +851,44 @@ Token Scanner::scanString(bool const _isUnicode)
 	char const quote = m_char;
 	advance();  // consume quote
 	LiteralScope literal(this, LITERAL_TYPE_STRING);
-	while (m_char != quote && !isSourcePastEndOfInput() && !isUnicodeLinebreak())
+	// for source location comments we allow multiline string literals
+	while (m_char != quote && !isSourcePastEndOfInput() && (!isUnicodeLinebreak() || m_kind == ScannerKind::SpecialComment))
 	{
 		char c = m_char;
 		advance();
-		if (c == '\\')
+
+		if (m_kind == ScannerKind::SpecialComment)
 		{
-			if (isSourcePastEndOfInput() || !scanEscape())
-				return setError(ScannerError::IllegalEscapeSequence);
+			if (c == '\\')
+				scanEscape();
```

Comment:
It would be more consistent with the other branch to treat unterminated `\` as `IllegalEscapeSequence`. Currently you continue, run `scanEscape()` and then error with `IllegalStringEndQuote` right after the loop.

It does work, but it's still weird to let `scanEscape()` run when we're at the end of input and `m_char` is `0`, which I guess is assumed to never actually be read. The function will actually take that `0` and put into the literal and that literal is still accessible via `currentLiteral()` even when we error out.

Also, you should either handle the return value from `scanEscape()` or assert that it's not supposed to fail:

```suggestion
			if (c == '\\')
			{
				if (isSourcePastEndOfInput())
					return setError(ScannerError::IllegalEscapeSequence);
				bool validEscape = scanEscape();
				solAssert(validEscape);
			}
```

Reply:
Right! Done that, tests now reflect a case where there's a dangling backslash `R"("test\)"` and a missing string end quote `R"("test)"`.